src/main/java/uk/gov/companieshouse/acsp/manage/users/controller/UserAcspMembership.java

```diff
@@ -29,10 +45,27 @@ public ResponseEntity<ResponseBodyPost> addAcspMember(
 
     @Override
     public ResponseEntity<AcspMembership> getAcspMembershipForAcspId(
-            @NotNull String xRequestId,
-            @Pattern(regexp = "^[a-zA-Z0-9]*$") String id
+           final String xRequestId,
+           final String id
     ) {
-        return null; // TODO(https://companieshouse.atlassian.net/browse/IDVA6-1146)
+        LOG.debug(String.format("%s: Retrieving id for member id (%s) ...", xRequestId, id));
+        if (Objects.isNull(id)){
+            LOG.error(String.format("%s: No member id was provided.", xRequestId));
+            throw new BadRequestRuntimeException(PLEASE_CHECK_THE_REQUEST_AND_TRY_AGAIN);
+        }
+
+        if ( !acspMembershipService.memberIdExists( id )){
```

Comment:
membership exists

Reply:
also you dont need to check if exists.

when you fetch the data using acspMembershipService.fetchAcspMembership(id) it will give you back empty optional. you cancheck against it.